guides/bulk.md

```diff
@@ -0,0 +1,271 @@
+let credentials = Credentials::Basic("admin".into(), "admin".into());
+let transport = TransportBuilder::new(SingleNodeConnectionPool::new(url))
+    .cert_validation(CertificateValidation::None)
+    .auth(credentials)
+    .build()?;
+let client = OpenSearch::new(transport);
+```
+
+Next, create an index named `movies` and another named `books` with the default settings:
+
+```rust
+let movies = "movies";
+let books = "books";
+client.indices().create(IndicesCreateParts::Index(movies)).send().await?;
+client.indices().create(IndicesCreateParts::Index(books)).send().await?;
+```
+
+## Bulk API
+
+The `bulk` API action allows you to perform document operations in a single request. The body of the request is an array of objects that contains the bulk operations and the target documents to index, create, update, or delete.
+
+### Indexing multiple documents
+
+The following code creates two documents in the `movies` index and one document in the `books` index:
+
+```rust
+client
+    .bulk(BulkParts::None)
+    .body(Vec::<JsonBody<_>>::from([
+        json!({ "index": { "_index": movies, "_id": "1" }}).into(),
+        json!({ "title": "Beauty and the Beast", "year": "1991" }).into(),
+        json!({ "index": { "_index": movies, "_id": "2" }}).into(),
+        json!({ "title": "Beauty and the Beast - Live Action", "year": "2017" }).into(),
+        json!({ "index": { "_index": books, "_id": "1" }}).into(),
+        json!({ "title": "The Lion King", "year": "1994" }).into(),
+    ]))
+    .send()
+    .await?;
+```
+
+As you can see, each bulk operation is comprised of two objects. The first object contains the operation type and the target document's `_index` and `_id`. The second object contains the document's data. As a result, the body of the request above contains six objects for three index actions.
+
+Alternatively, the `bulk` method can accept an array of hashes where each hash represents a single operation. The following code is equivalent to the previous example:
+
+```rust
+let mut ops = BulkOperations::new();
+ops.push(
+    BulkOperation::index(json!({
+        "title": "Beauty and the Beast",
+        "year": "1991"
+    }))
+    .index(movies)
+    .id("1"),
+)?;
+ops.push(
+    BulkOperation::index(json!({
+        "title": "Beauty and the Beast - Live Action",
+        "year": "2017"
+    }))
+    .index(movies)
+    .id("2"),
+)?;
+ops.push(
+    BulkOperation::index(json!({
+        "title": "The Lion King",
+        "year": "1994"
+    }))
+    .index(books)
+    .id("1"),
+)?;
+
+client.bulk(BulkParts::None).body(vec![ops]).send().await?;
+```
+
+We will use this format for the rest of the examples in this guide.
+
+### Creating multiple documents
+
+Similarly, instead of calling the `create` method for each document, you can use the `bulk` API to create multiple documents in a single request. The following code creates three documents in the `movies` index and one in the `books` index:
```

Comment:
Let's swap this to the `index` operation.

```suggestion
### Indexing multiple documents

Similarly, instead of calling the `index` method for each document, you can use the `bulk` API to index multiple documents in a single request. The following code indexes three documents in the `movies` index and one in the `books` index:
```